asciidoc/settings/settings.adoc

```diff
@@ -50,6 +50,7 @@ So if there is a setting in the resolver, it always wins against all other setti
 |validate|Indicates if ivy files should be validated against ivy.xsd or not.|No, defaults to true
 |useRemoteConfig|true to configure ivyrep and ibiblio resolver from a remote configuration file (updated with changes in those repository structure if any) (*__since 1.2__*)|No, defaults to false
 |httpRequestMethod|specifies the HTTP method to use to retrieve information about an URL. Possible values are 'GET' and 'HEAD'. This setting can be used to solve problems with firewalls and proxies. (*__since 2.0__*)|No, defaults to 'HEAD'
+|preferPreemptiveAuth|true to use link:https://hc.apache.org/httpcomponents-client-4.5.x/tutorial/html/authentication.html[preemptive authentication] whenever possible (only supported with link:https://hc.apache.org/httpcomponents-client-4.5.x[HttpClient] as URL Handler) (*__since 4.5__*)|No, defaults to false
```

Comment:
Why not just preemptiveAuth? That is more natural then prefer, also see other comment as also with HttpURLConnection it is possible.

Reply:
There are two classes extending `AbstractURLHandler`: `HttpClientHandler` in which I make use of `preferPreemptiveAuth`, and `BasicURLHandler` in which I ignore it. 

This means, this setting isn't necessarily taken into account. It's just a **preference**, which is followed or not, depending on what implementation of `AbstractURLHandler` you use.

Now when grepping for `HttpURLConnection`, it looks like you mean the `BasicURLHandler`, which makes use of it. It might be possible to implement there as well, but not done yet, because we don't need it and also because the author of https://issues.apache.org/jira/browse/IVY-1280 is using HttpClient as well.

Reply:
That is exactly my point, that we are not using it yet or someone else doesn't request it, doesn't mean we have to name it as preference. The default usage is still use the Java JDK HttpURLConnection. We can just call it preemptiveAuth and decide to not implement it for other URL handlers and let it act as noop basically for time being until implemented. Having it named after its purpose as API setting make imho more sense, but that might be just a taste.

Reply:
I was thinking that if someone implements it for `BasicURLHandler` as well some day, they could rename the settings. But that would be a breaking change, so I think you're right and I will rename it now.